megamek/src/megamek/client/ui/swing/tileset/TilesetManager.java

```diff
@@ -0,0 +1,556 @@
+        }
+        PreferenceManager.getClientPreferences().addPreferenceChangeListener(this);
+        GUIPreferences.getInstance().addPreferenceChangeListener(this);
+    }
+
+    /** React to changes in the settings. */
+    public void preferenceChange(PreferenceChangeEvent e) {
+        // A new Hex Tileset has been selected
+        if (e.getName().equals(IClientPreferences.MAP_TILESET)) {
+            HexTileset hts = new HexTileset(boardview.game);
+            try {
+                hexTileset.incDepth = 0;
+                hts.loadFromFile((String) e.getNewValue());
+                hexTileset = hts;
+                boardview.clearHexImageCache();
+            } catch (IOException ex) {
+                return;
+            }
+        }
+        
+        // The setting to show damage decals and smoke has changed
+        if (e.getName().equals(GUIPreferences.SHOW_DAMAGE_DECAL)) {
+            reset();
+        }
+    }
+
+    /** Retrieve an icon for the unit (used in the Unit Overview). */
+    public Image iconFor(Entity entity) {
+        EntityImage entityImage = getFromCache(entity, -1);
+        if (entityImage == null) {
+            System.err.println("Unable to load icon for entity: " + entity.getShortNameRaw()); //$NON-NLS-1$
+            Image generic = getGenericImage(entity, -1, mechTileset);
+            return (generic != null) ? ImageUtil.getScaledImage(generic, 56, 48) : null;
+        }
+        return entityImage.getIcon();
+    }
+
+    /** Retrieve a wreck icon for the unit. */
+    public Image wreckMarkerFor(Entity entity, int secondaryPos) {
+        EntityImage entityImage = getFromCache(entity, secondaryPos);
+        if (entityImage == null) {
+            System.out.println("Unable to load wreckMarker image for entity: " + entity.getShortNameRaw()); //$NON-NLS-1$
+            return getGenericImage(entity, -1, wreckTileset);
+        }
+        return entityImage.getWreckFacing(entity.getFacing());
+    }
+
+    /** Retrieve an image for the unit. */
+    public Image imageFor(Entity entity) {
+        return imageFor(entity, -1);
+    }
+
+    /** Retrieve an image for the unit. */
+    public Image imageFor(Entity entity, int secondaryPos) {
+        // mechs look like they're facing their secondary facing
+        // (except QuadVees, which are using turrets instead of torso twists
+        if (((entity instanceof Mech) || (entity instanceof Protomech))
+                && !(entity instanceof QuadVee)) {
+            return imageFor(entity, entity.getSecondaryFacing(), secondaryPos);
+        }
+        return imageFor(entity, entity.getFacing(), secondaryPos);
+    }
+
+    /** Retrieve an image for the unit. */
+    public Image imageFor(Entity entity, int facing, int secondaryPos) {
+        EntityImage entityImage = getFromCache(entity, secondaryPos);
+        if (entityImage == null) {
+            System.err.println("Unable to load image for entity: " + entity.getShortNameRaw()); //$NON-NLS-1$
```

Comment:
I'm going to stop calling out system.println and just say that it would be useful to go through this file and replace them with calls to actual logging.